packages/amplify-codegen-model-plugin/src/languages/typescript-declaration-block.ts

```diff
@@ -0,0 +1,267 @@
+    }
+    this._methods.push({
+      name,
+      returnType,
+      implmentation,
+      args,
+      flags,
+      access,
+      comment: transformComment(comment),
+    });
+  }
+
+  public get string(): string {
+    switch (this._kind) {
+      case 'interface':
+        return this.generateInterface();
+      case 'class':
+        return this.generateClass();
+      case 'enum':
+        return this.generateEnum();
+    }
+  }
+
+  protected generateEnum(): string {
+    if (this._kind !== 'enum') {
+      throw new Error(`generateEnum called for non enum kind(${this._kind})`);
+    }
+    const header = ['export', 'enum', this._name].join(' ');
+    const body = Object.entries(this._enumValues)
+      .map(([name, value]) => `${name} = "${value}"`)
+      .join(',\n');
+    return [`${header} {`, indentMultiline(body), '}'].join('\n');
+  }
+
+  protected generateClass(): string {
+    const header: string[] = [
+      this._flags.shouldExport ? 'export' : '',
+      this._flags.isDeclaration ? 'declare' : '',
+      'class',
+      this._name,
+      '{',
+    ];
+    if (this._extends.length) {
+      header.push(['extends', this._extends.join(', ')].join(' '));
+    }
+    const body: string[] = [this.generateProperties(), this.generateMethods()];
+
+    return [`${header.filter(h => h).join(' ')}`, indentMultiline(body.join('\n')), '}'].join('\n');
+  }
+
+  protected generateProperties(): string {
+    const props = this._properties.map(prop => {
+      const result: string[] = [];
+      if (prop.access !== 'DEFAULT') {
+        result.push(prop.access);
+      }
+      if (prop.flags && prop.flags.readonly) {
+        result.push('readonly');
+      }
+
+      result.push(this.generatePropertyName(prop));
+
+      if (prop.value) {
+        result.push(` = ${prop.type};`);
+      }
+
+      return result.join(' ');
+    });
+    return props.map(propDeclaration => `${propDeclaration};`).join('\n');
+  }
+
+  protected generateMethods(): string {
+    const methods: string[] = this._methods.map(method => {
+      const methodAccessAndName = [];
+      if (method.access !== 'DEFAULT') {
+        methodAccessAndName.push(access);
+      }
+      if (method.flags.static) {
+        methodAccessAndName.push('static');
+      }
+      methodAccessAndName.push(method.name);
+
+      const args = method.args
+        .map(arg => {
+          return `${arg.name}${arg.type ? `: ${arg.type}` : ''}`;
+        })
+        .join(', ');
+
+      const returnType = method.returnType ? `: ${method.returnType}` : '';
+      const methodHeaderStr = `${methodAccessAndName.join(' ')}(${args})${returnType}`;
+
+      if (this._flags.isDeclaration) {
+        return `${methodHeaderStr};`;
+      }
+      return [`${methodHeaderStr}`, '{', method.implmentation ? indentMultiline(method.implmentation) : '', '}'].join('\n');
+    });
+    return methods.join('\n');
+  }
+
+  protected generateInterface(): string {
```

Comment:
Not implemented "yet"? or we don't support interfaces? Based on the answer we should update the text here.